Cauldron/Assets/Scripts/TutorialStorage.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using Save;
+using UnityEngine;
+
+public enum TutorialKeys
+{
+    TUTORIAL_BOOK_OPENED,
+    BOOK_AUTOCOOKING_OPENED,
+    TUTORIAL_VISITOR,
+    TUTORIAL_CHANGE_SCALE,
+    TUTORIAL_POTION_DENIED,
+    TUTORIAL_RECIPE_HINTS 
+}
+
+[Serializable]
+public class Tutorials
+{
+    public bool TUTORIAL_BOOK_OPENED = false;
+    public bool BOOK_AUTOCOOKING_OPENED = false;
+    public bool TUTORIAL_VISITOR = false;
+    public bool TUTORIAL_CHANGE_SCALE = false;
+    public bool TUTORIAL_POTION_DENIED = false;
+    public bool TUTORIAL_RECIPE_HINTS = false;
+
+    public void SetValue(TutorialKeys key, bool value)
```

Comment:
We can reduce the code by having a HashSet of enum values instead of a set of bools. When setting, we can add a corresponding enum value to the set. When getting, we check if Set contains the given enum value.
This approach would support additional tutorial flags as well, I'm not sure we can do it with your fixed set of bools.

Or we can further optimize the storage by using a bit mask (Flags). I believe endings in the EndingsProvider use this approach. You can try it if you want.

Reply:
Btw, we never set tutorials to false.